_studio/shared/src/libmfx_core_vaapi.cpp

```diff
@@ -612,7 +612,9 @@ VAAPIVideoCORE::AllocFrames(
         } else
         {
             // external allocator
-            if (m_bSetExtFrameAlloc && request->Info.FourCC != MFX_FOURCC_P8)
+            if (m_bSetExtFrameAlloc &&
+                request->Info.FourCC != MFX_FOURCC_P8 &&
```

Comment:
let's do "(request->Type & MFX_MEMTYPE_EXTERNAL_FRAME" instead of "!(request->Type & MFX_MEMTYPE_INTERNAL_FRAME))"

Reply:
done